jetstream/util.py

```diff
@@ -20,3 +25,16 @@ def inclusive_date_range(start_date, end_date):
     """Generator for a range of dates, includes end_date."""
     for n in range(int((end_date - start_date).days) + 1):
         yield start_date + timedelta(n)
+
+
+def retry_get(session, url, max_retries, verify=None, headers=None) -> Any:
+    for _i in range(max_retries):
+        try:
+            blob = session.get(url, verify=verify, headers=headers).json()
+            break
+        except Exception:
+            logger.info(f"Error fetching from {url}. Retrying...")
+            time.sleep(1)
+    else:
+        raise Exception(f"Too many retries for {url}")
+    return blob
```

Comment:
We can reuse the retry logic used for the Experimenter API also for the Argo API. Not sure if just having a `retry_get` function is sufficient or if we'd want to have a separate class similar to what is described in https://findwork.dev/blog/advanced-usage-python-requests-timeouts-retries-hooks?

Reply:
Oh. the `urllib3.util.retry` strategy is what I had in mind, but this should also work well!